### PR TITLE
Feature/update upgrade script

### PR DESCRIPTION
**Pull request by invocamanman**
- Upgrade script hackmd updated

### CODE REVIEW

**Review by ignasirv — APPROVED** [review]
💯

File: deployment/README.md [hunk 1]
```diff
@@ -46,15 +48,17 @@ npm run verify:ZkEVM:goerli
...
 - `PolygonZkEVMBridgeMock`:Boolean, Wheather the PolygonZkEVMBridge will be mock or not ( the mock version has a ether limitation on deposits)
 - `admin`:address, Admin address
 - `trustedAggregator`:address, Trusted aggregator address
+- `minDelayTimelock`: number, minimum timelock delay,
+- `timelockAddress`: address, Timelock owner address
 
 ## Notes
 
-- `gensis.json` has been generated using the tool: `https://github.com/0xPolygonHermez/zkevm-commonjs/blob/main/tools/fill-genesis/create-genesis.js` using as generator file: `genesis-gen.json`
+- `gensis.json` has been generated using the tool: `src/create-genesis.js` using as generator file: `genesis-gen.json`
```
Comment (krlosMata) [inline deployment/README.md:64]:
```suggestion
- `genesis.json` has been generated using the tool: `src/create-genesis.js` using as generator file: `genesis-gen.json`
```

File: upgrade/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+In project root, copy the `${network}.json` of the deployment that you want to upgrade and copy it on the `./.openzeppelin`
+e.g. `cp deployments/${network}_$(date +%s)/${network}.json ./.openzeppelin`
+
+if the deployment was deployed without a timelock you can use the `simpleUpgradeScript.js`:
+
+- Update the `proxyPolygonAddress` with the proxy address you want to update
+- Update the `polygonZkEVMFactory` with the new implementation contract you want to upgrade
+- Run the script
+
+Otherwise, inc ase of timelock use `timeLockUpgrade.js`
```
Comment (krlosMata) [inline upgrade/README.md:26]:
```suggestion
Otherwise, in case of timelock use `timeLockUpgrade.js`
```

File: upgrade/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+- Update the `proxyPolygonAddress` with the proxy address you want to update
+- Update the `polygonZkEVMFactory` with the new implementation contract you want to upgrade
+- Update the `minDelay` with the delay that you want to use
+- Run the script
+- Now the necessary transactions to interact with the timelock are printed in the screen `schedule` and `execute`
+- With the owner of the timelock ( multisig or account), send the data printed by `schedule` to the `Timelock` contract.
```
Comment (krlosMata) [inline upgrade/README.md:33]:
```suggestion
- With the owner of the timelock (multisig or account), send the data printed by `schedule` to the `Timelock` contract.
```

File: upgrade/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+- Once the necessary timeout has pass, with the same account you can now send the data printed by `execute` to the `Timelock` contract and the contracts will be upgraded.
```
Comment (krlosMata) [inline upgrade/README.md:34]:
```suggestion
- Once the necessary timeout has expired, with the same account you can now send the data printed by `execute` to the `Timelock` contract and the contracts will be upgraded.
```

File: upgrade/simpleUpgradeScript.js [hunk 5]
```diff
@@ -27,24 +27,15 @@ async function main() {
     // compìle contracts
     await hre.run('compile');
 
-    const polygonZkEVMProxyAddress = '0xfefefefefefefefefefefefee';
-    const polygonZkEVMFactory = await ethers.getContractFactory('PolygonZkEVMMock');
+    const proxyPolygonAddress = '0xfefefefefefefefefefefefee';
```
Comment (krlosMata) [inline upgrade/simpleUpgradeScript.js:30]: This parameter should be loaded by `cli` or `input-upgrade.json`

File: upgrade/timeLockUpgrade.js [hunk 6]
```diff
@@ -27,23 +27,26 @@ async function main() {
     // compìle contracts
```
Comment (krlosMata) [inline upgrade/timeLockUpgrade.js:27]: Output with upgrade information should be provided as well in json format

**Review by krlosMata — CHANGES_REQUESTED** [review]

File: upgrade/timeLockUpgrade.js [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+            [
+                operation.target,
+                operation.value,
+                operation.data,
+                operation.predecessor,
+                operation.salt,
+                minDelay,
+            ],
+        );
+        // Executre operation
```
Comment (krlosMata) [inline upgrade/timeLockUpgrade.js:84]:
```suggestion
        // Execute operation
```

File: upgrade/timeLockUpgrade.js [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+        [deployer] = (await ethers.getSigners());
+    }
+    // compìle contracts
+    await hre.run('compile');
+
+    const proxyAdmin = await upgrades.admin.getInstance();
+    const output = [];
+
+    // Upgrade zkevm
+    for (const upgrade of upgradeParameters.upgrades) {
```
Comment (krlosMata) [inline upgrade/timeLockUpgrade.js:44]: I would check in advance `upgrade.contractName` existence

**Review by krlosMata — CHANGES_REQUESTED** [review]

**Review by krlosMata — APPROVED** [review]
:smile_cat:

**Pull request merged**